shared/domains/chelonia/utils.js

```diff
@@ -790,3 +791,14 @@ export const clearObject = (o: Object) => {
 export const reactiveClearObject = (o: Object, fn: (o: Object, k: string | number) => any) => {
   Object.keys(o).forEach((k) => fn(o, k))
 }
+
+export const checkCanBeGarbageCollected = function (id: string) {
+  const rootState = sbp(this.config.stateSelector)
+  return (
+    // Check persistent references
+    (!has(rootState.contracts, id) || !has(rootState.contracts[id], 'references')) &&
+    // Check ephemeral references
+    !has(this.ephemeralReferenceCount, id)) &&
+    // Check foreign keys (i.e., that no keys are being watched)
```

Comment:
Please clarify a bit more in this comment: are we checking whether this `id` is watching other contracts? Or are we checking whether other contracts are watching this `id`? Or both?

Reply:
I only moved this code from where it was, but I can clarify.